runtime/memory.c

```diff
@@ -277,6 +279,8 @@ char *caml_alloc_for_heap (asize_t request)
     mem += sizeof (heap_chunk_head);
     Chunk_size (mem) = request;
     Chunk_block (mem) = block;
+    Chunk_redarken_start(mem) = (value*)(mem + request);
```

Comment:
This initialization is not quite the same as that a few lines above: here you're taking `mem + Chunk_size(mem)` while previously you took `mem + Chunk_size(mem) + sizeof(heap_chunk_head)`.  It probably makes no difference but feels like it isn't intented.

Reply:
Indeed this was an oversight. These have been changed to `(value*)-1` in the most recent commit.